### ISSUE TITLE
Error "Re-enable cpu cache" while using PSRAM in WROVER kit (IDFGH-4419)

### ISSUE THREAD

**Issue by ansee7221** [espressif/esp-idf#6250]
## Environment
- Development Kit:      [ESP32-Wrover-Kit]
- Kit version (for WroverKit): [v1|v2|v3|v4]
- Module or chip used:  [ESP32-WROVER-B]
- IDF version (run ``git describe --tags`` to find it):
    //v4.3-dev-1521-g0289d1c
- Build System:         [idf.py]
- Compiler version (run ``xtensa-esp32-elf-gcc --version`` to find it):
    // crosstool-NG esp-2020r3
- Operating System:     [Linux]
- (Windows only) environment type: [MSYS2 mingw32|ESP Command Prompt|Plain Command Prompt|PowerShell].
- Using an IDE?: [No]
- Power Supply:         [USB]

## Problem Description
I set up the WROVER kit to use PSRAM and it reboots with a "Re-enable cpu cache" error.

//Detailed problem description goes here.

### Steps to reproduce

// If possible, attach a picture of your setup/wiring here.

### Code to reproduce this issue

```cpp
// the code should be wrapped in the ```cpp tag so that it will be displayed better.
#include "esp_log.h"

void app_main()
{
	if (nvs_flash_init() != ESP_OK) {
		print_error("nvs_flash_init failed");
	}
	//gpio_reset_pin(BLINK_GPIO);
	//gpio_set_direction(BLINK_GPIO, GPIO_MODE_OUTPUT);

	pthread_cond_init(&cv, NULL);

	print_macro_info();
	initialise_wifi();
	initialize_console();

	// Register commands	   
	esp_console_register_help_command();
	register_ctt();

#ifdef EXTERNAL_SPRAM_USE

	// Create the task without using any dynamic memory allocation.
	xHandle = xTaskCreateStatic(	  
	server_main,	 // Function that implements the task.		
	"server_main", // Text name for the task.	   
	STACK_SIZE,	  // Stack size in bytes, not words.	  
	NULL, 		 // Parameter passed into the task. 	 
	5, 			// Priority at which the task is created.	   
	xStack,		  // Array to use as the task's stack.		
	&xTaskBuffer); // Variable to hold the task's data structure.

	if (xHandle == NULL) {
		print_error("task create failed (server_main)");
	} 

	xHandleConsole = xTaskCreateStatic(
	console_task, "console", 
	STACK_SIZE_CONSOLE, 
	NULL, 
	configMAX_PRIORITIES - 2, 
	xStackConsole, 
	&xTaskBufferConsole); 

	if (xHandleConsole == NULL) {
		print_error("task create failed (console)");
	}

#else

	// Create the task without using any dynamic memory allocation.
 	if ( xTaskCreate(&server_main, "server_main", 
		//17*1024, 
		1024*18, 
		NULL, 5, NULL) != pdPASS ) {
		print_error("task create failed(server_main)");
	}
 
 	if ( xTaskCreate(&console_task, "console", 
		//1024*3, 
		1024*2, 
		NULL, configMAX_PRIORITIES - 2, NULL) != pdPASS ) {
		print_error("task create failed(console)");
	} 

#endif
}

```
// If your code is longer than 30 lines, [GIST](https://gist.github.com) is preferred.

## Debug Logs

ets Jun  8 2016 00:22:57
rst:0x1 (POWERON_RESET),boot:0x1e (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:4
load:0x3fff0034,len:7200
ho 0 tail 12 room 4
load:0x40078000,len:15316
load:0x40080400,len:3108
entry 0x40080600
I (30) boot: ESP-IDF v4.3-dev-1521-g0289d1c-dirty 2nd stage bootloader
I (31) boot: compile time 16:11:25
I (32) boot: chip revision: 1
I (35) boot_comm: chip revision: 1, min. bootloader chip revision: 0
I (42) boot.esp32: SPI Speed      : 40MHz
I (47) boot.esp32: SPI Mode       : DIO
I (51) boot.esp32: SPI Flash Size : 4MB
I (56) boot: Enabling RNG early entropy source...
I (61) boot: Partition Table:
I (65) boot: ## Label            Usage          Type ST Offset   Length
I (72) boot:  0 nvs              WiFi data        01 02 00009000 00006000
I (80) boot:  1 phy_init         RF data          01 01 0000f000 00001000
I (87) boot:  2 factory          factory app      00 00 00010000 00140000
I (95) boot: End of partition table
I (99) boot_comm: chip revision: 1, min. application chip revision: 0
I (106) esp_image: segment 0: paddr=0x00010020 vaddr=0x3f400020 size=0x2d390 (185232) map
I (192) esp_image: segment 1: paddr=0x0003d3b8 vaddr=0x3ffb0000 size=0x02c60 ( 11360) load
I (198) esp_image: segment 2: paddr=0x00040020 vaddr=0x400d0020 size=0xc37f4 (800756) map
I (533) esp_image: segment 3: paddr=0x0010381c vaddr=0x3ffb2c60 size=0x01e74 (  7796) load
I (537) esp_image: segment 4: paddr=0x00105698 vaddr=0x40080000 size=0x00404 (  1028) load
I (540) esp_image: segment 5: paddr=0x00105aa4 vaddr=0x40080404 size=0x1e1f0 (123376) load
I (625) boot: Loaded app from partition at offset 0x10000
I (625) boot: Disabling RNG early entropy source...
I (637) psram: This chip is ESP32-D0WD
I (637) spiram: Found 64MBit SPI RAM device
I (637) spiram: SPI RAM mode: flash 40m sram 40m
I (642) spiram: PSRAM initialized, cache is in low/high (2-core) mode.
I (649) cpu_start: Pro cpu up.
I (653) cpu_start: Starting app cpu, entry point is 0x40081880
I (0) cpu_start: App cpu up.
I (1523) spiram: SPI SRAM memory test OK
I (1532) cpu_start: Pro cpu start user code
I (1532) cpu_start: cpu freq: 160000000
I (1532) cpu_start: Application information:
I (1535) cpu_start: Project name:     KD_iotivity_server
I (1541) cpu_start: App version:      2.2.0-28-g9fc5ba6-dirty
I (1547) cpu_start: Compile time:     Dec 11 2020 16:12:09
I (1553) cpu_start: ELF file SHA256:  003d9eb7016a05a0...
I (1559) cpu_start: ESP-IDF:          v4.3-dev-1521-g0289d1c-dirty
I (1567) heap_init: Initializing. RAM available for dynamic allocation:
I (1574) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (1580) heap_init: At 3FFBBE70 len 00024190 (144 KiB): DRAM
I (1586) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (1592) heap_init: At 3FFE4350 len 00013CB0 (79 KiB): D/IRAM
I (1599) heap_init: At 4009E5F4 len 00001A0C (6 KiB): IRAM
I (1605) spiram: Adding pool of 4084K of external SPI memory to heap allocator
I (1614) esp_apptrace: Initialized TRAX on CPU0
I (1619) spi_flash: detected chip: generic
I (1623) spi_flash: flash io: dio
I (1629) cpu_start: Starting scheduler on PRO CPU.
I (537) esp_apptrace: Initialized TRAX on CPU1
I (0) cpu_start: Starting scheduler on APP CPU.

****************************************
OC_DEBUG defined!
APP_DEBUG defined!
OC_CLIENT defined!
OC_SERVER defined!
OC_IPV4 not defined!
ENABLE_LIGHT not defined!
OC_SECURITY defined!
OC_DYNAMIC_ALLOCATION defined!
OC_BLOCK_WISE defined!
OC_PKI defined!
OC_TCP not defined!

****************************************
I (1873) wifi:wifi driver task: 3ffc6158, prio:23, stack:6656, core=0
I (1873) system_api: Base MAC address is not set
I (1873) system_api: read default base MAC address from EFUSE
I (1923) wifi:wifi firmware version: a94a000
I (1923) wifi:wifi certification version: v7.0
I (1923) wifi:config NVS flash: enabled
I (1923) wifi:config nano formating: disabled
I (1933) wifi:Init data frame dynamic rx buffer num: 32
I (1933) wifi:Init management frame dynamic rx buffer num: 32
I (1943) wifi:Init management short buffer num: 32
I (1943) wifi:Init static tx buffer num: 16
I (1953) wifi:Init tx cache buffer num: 32
I (1953) wifi:Init static rx buffer size: 1600
I (1953) wifi:Init static rx buffer num: 10
I (1963) wifi:Init dynamic rx buffer num: 32
I (1963) wifi_init: rx ba win: 6
I (1963) wifi_init: tcpip mbox: 32
I (1973) wifi_init: udp mbox: 6
I (1973) wifi_init: tcp mbox: 6
I (1983) wifi_init: tcp tx win: 5744
I (1983) wifi_init: tcp rx win: 5744
I (1983) wifi_init: tcp mss: 1440
I (1993) wifi_init: WiFi IRAM OP enabled
I (1993) wifi_init: WiFi RX IRAM OP enabled
I (2003) wifi_init: LWIP IRAM OP enabled
I (2113) phy: phy_version: 4500, 0cd6843, Sep 17 202 0I (2123) iotivity server: iotivity server task started

This is an example of ESP-IDF console component.
Type 'help' to get the list of commands.
Use UP/DOWN arrows to navigate through command history.
Press TAB when typing command name to auto-complete.
I (2603) wifi:new:<4,0>, old:<1,0>, ap:<255,255>, sta:<4,0>, prof:1
I (3643) wifi:state: init -> auth (b0)
I (3663) wifi:state: auth -> assoc (0)
I (3673) wifi:state: assoc -> run (10)
I (3703) wifi:connected with TP-Link_FC80, aid = 3, channel 4, BW20, bssid = 84:d8:1b:35:fc:7f
I (3703) wifi:security: WPA2-PSK, phy: bgn, rssi: -27
I (3703) wifi:pm start, type: 1

W (3723) wifi:<ba-add>idx:0 (ifx:0, 84:d8:1b:35:fc:7f), tid:5, ssn:2, winSize:64
I (3763) wifi:AP's beacon interval = 102400 us, DTIM period = 1
I (4363) esp_netif_handlers: iotivity server: sta ip: 192.168.0.150, mask: 255.255.255.0, gw: 192.168.0.1
W (4483) wifi:<ba-add>idx:1 (ifx:0, 84:d8:1b:35:fc:7f), tid:0, ssn:0, winSize:64
I (4863) iotivity server: got IPv6 addr:FE80::AE67:B2FF:FE71:617C
         introspection via header file
app_init()
Register Resource with local path "/power"
          number of Resource Types: 1
        Resource Type: "oic.r.switch.binary"
          Default OCF Interface: "oic.if.a"
Register Resource with local path "/temperature"
          number of Resource Types: 1
        Resource Type: "oic.r.temperature"
          Default OCF Interface: "oic.if.a"
WARNING: ../adapter/src/storage.c <oc_storage_read:65>: oc_storage_read cannot nvs_get_blob  u_ids_0: 4354
WARNING: ../adapter/src/storage.c <oc_storage_read:65>: oc_storage_read cannot nvs_get_blob  pstat_0: 4354
Successfully installed PKI certificate

ets Jun  8 2016 00:22:57
rst:0x8 (TG1WDT_SYS_RESET),boot:0x1e (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:4
load:0x3fff0034,len:7200
ho 0 tail 12 room 4
load:0x40078000,len:15316
load:0x40080400,len:3108
entry 0x40080600
I (61) boot: ESP-IDF v4.3-dev-1521-g0289d1c-dirty 2nd stage bootloader
I (62) boot: compile time 16:11:25
I (63) boot: chip revision: 1
I (67) boot_comm: chip revision: 1, min. bootloader chip revision: 0
I (74) boot.esp32: SPI Speed      : 40MHz
I (79) boot.esp32: SPI Mode       : DIO
I (83) boot.esp32: SPI Flash Size : 4MB
W (88) boot.esp32: PRO CPU has been reset by WDT.
**W (93) boot.esp32: WDT reset info: PRO CPU PC=0x40085ad0
W (99) boot.esp32: WDT reset info: APP CPU PC=0x4009201a**
I (105) boot: Enabling RNG early entropy source...
I (111) boot: Partition Table:
I (114) boot: ## Label            Usage          Type ST Offset   Length
I (122) boot:  0 nvs              WiFi data        01 02 00009000 00006000
I (129) boot:  1 phy_init         RF data          01 01 0000f000 00001000
I (137) boot:  2 factory          factory app      00 00 00010000 00140000
I (144) boot: End of partition table
I (148) boot_comm: chip revision: 1, min. application chip revision: 0
I (156) esp_image: segment 0: paddr=0x00010020 vaddr=0x3f400020 size=0x2d390 (185232) map
I (242) esp_image: segment 1: paddr=0x0003d3b8 vaddr=0x3ffb0000 size=0x02c60 ( 11360) load
I (248) esp_image: segment 2: paddr=0x00040020 vaddr=0x400d0020 size=0xc37f4 (800756) map
I (583) esp_image: segment 3: paddr=0x0010381c vaddr=0x3ffb2c60 size=0x01e74 (  7796) load
I (587) esp_image: segment 4: paddr=0x00105698 vaddr=0x40080000 size=0x00404 (  1028) load
I (590) esp_image: segment 5: paddr=0x00105aa4 vaddr=0x40080404 size=0x1e1f0 (123376) load
I (674) boot: Loaded app from partition at offset 0x10000
I (675) boot: Disabling RNG early entropy source...
I (686) psram: This chip is ESP32-D0WD
I (687) spiram: Found 64MBit SPI RAM device
I (687) spiram: SPI RAM mode: flash 40m sram 40m
I (692) spiram: PSRAM initialized, cache is in low/high (2-core) mode.
I (699) cpu_start: Pro cpu up.
I (703) cpu_start: Starting app cpu, entry point is 0x40081880
I (0) cpu_start: App cpu up.
I (1572) spiram: SPI SRAM memory test OK
I (1581) cpu_start: Pro cpu start user code
I (1581) cpu_start: cpu freq: 160000000
I (1581) cpu_start: Application information:
I (1584) cpu_start: Project name:     KD_iotivity_server
I (1590) cpu_start: App version:      2.2.0-28-g9fc5ba6-dirty
I (1597) cpu_start: Compile time:     Dec 11 2020 16:12:09
I (1603) cpu_start: ELF file SHA256:  003d9eb7016a05a0...
I (1609) cpu_start: ESP-IDF:          v4.3-dev-1521-g0289d1c-dirty
I (1616) heap_init: Initializing. RAM available for dynamic allocation:
I (1623) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (1629) heap_init: At 3FFBBE70 len 00024190 (144 KiB): DRAM
I (1636) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (1642) heap_init: At 3FFE4350 len 00013CB0 (79 KiB): D/IRAM
I (1649) heap_init: At 4009E5F4 len 00001A0C (6 KiB): IRAM
I (1655) spiram: Adding pool of 4084K of external SPI memory to heap allocator
I (1664) esp_apptrace: Initialized TRAX on CPU0
I (1668) spi_flash: detected chip: generic
I (1673) spi_flash: flash io: dio
I (1679) cpu_start: Starting scheduler on PRO CPU.
I (537) esp_apptrace: Initialized TRAX on CPU1
I (0) cpu_start: Starting scheduler on APP CPU.

****************************************
OC_DEBUG defined!
APP_DEBUG defined!
OC_CLIENT defined!
OC_SERVER defined!
OC_IPV4 not defined!
ENABLE_LIGHT not defined!
OC_SECURITY defined!
OC_DYNAMIC_ALLOCATION defined!
OC_BLOCK_WISE defined!
OC_PKI defined!
OC_TCP not defined!

****************************************
I (1962) wifi:wifi driver task: 3ffc6258, prio:23, stack:6656, core=0
I (1962) system_api: Base MAC address is not set
I (1962) system_api: read default base MAC address from EFUSE
I (2012) wifi:wifi firmware version: a94a000
I (2012) wifi:wifi certification version: v7.0
I (2012) wifi:config NVS flash: enabled
I (2012) wifi:config nano formating: disabled
I (2022) wifi:Init data frame dynamic rx buffer num: 32
I (2022) wifi:Init management frame dynamic rx buffer num: 32
I (2032) wifi:Init management short buffer num: 32
I (2032) wifi:Init static tx buffer num: 16
I (2032) wifi:Init tx cache buffer num: 32
I (2042) wifi:Init static rx buffer size: 1600
I (2042) wifi:Init static rx buffer num: 10
I (2052) wifi:Init dynamic rx buffer num: 32
I (2052) wifi_init: rx ba win: 6
I (2052) wifi_init: tcpip mbox: 32
I (2062) wifi_init: udp mbox: 6
I (2062) wifi_init: tcp mbox: 6
I (2072) wifi_init: tcp tx win: 5744
I (2072) wifi_init: tcp rx win: 5744
I (2072) wifi_init: tcp mss: 1440
I (2082) wifi_init: WiFi IRAM OP enabled
I (2082) wifi_init: WiFi RX IRAM OP enabled
I (2092) wifi_init: LWIP IRAM OP enabled
I (2202) phy: phy_version: 4500, 0cd6843, Sep 17 202A0I (2212) iotivity server: iotivity server task started

This is an example of ESP-IDF console component.
Type 'help' to get the list of commands.
Use UP/DOWN arrows to navigate through command history.
Press TAB when typing command name to auto-complete.
I (2692) wifi:new:<4,0>, old:<1,0>, ap:<255,255>, sta:<4,0>, prof:1
I (3732) wifi:state: init -> auth (b0)
I (3752) wifi:state: auth -> assoc (0)
I (3752) wifi:state: assoc -> init (3c0)
I (3752) wifi:new:<4,0>, old:<4,0>, ap:<255,255>, sta:<4,0>, prof:1
I (5932) wifi:new:<4,0>, old:<4,0>, ap:<255,255>, sta:<4,0>, prof:1
I (5932) wifi:state: init -> auth (b0)
I (5942) wifi:state: auth -> assoc (0)
I (5952) wifi:state: assoc -> run (10)
I (5972) wifi:connected with TP-Link_FC80, aid = 3, channel 4, BW20, bssid = 84:d8:1b:35:fc:7f
I (5982) wifi:security: WPA2-PSK, phy: bgn, rssi: -29
I (5982) wifi:pm start, type: 1

W (6002) wifi:<ba-add>idx:0 (ifx:0, 84:d8:1b:35:fc:7f), tid:5, ssn:2, winSize:64
I (6032) wifi:AP's beacon interval = 102400 us, DTIM period = 1
I (6952) esp_netif_handlers: iotivity server: sta ip: 192.168.0.150, mask: 255.255.255.0, gw: 192.168.0.1
I (7952) iotivity server: got IPv6 addr:FE80::AE67:B2FF:FE71:617C
         introspection via header file
app_init()
Register Resource with local path "/power"
          number of Resource Types: 1
        Resource Type: "oic.r.switch.binary"
          Default OCF Interface: "oic.if.a"
Register Resource with local path "/temperature"
          number of Resource Types: 1
        Resource Type: "oic.r.temperature"
          Default OCF Interface: "oic.if.a"
WARNING: ../adapter/src/storage.c <oc_storage_read:65>: oc_storage_read cannot nvs_get_blob  pstat_0: 4354
Successfully installed PKI certificate
Successfully installed intermediate CA certificate
**Re-enable cpu cache.
Guru Meditation Error: Core  0 panic'ed (Cache disabled but cached memory region accessed).**

Core  0 register dump:
PC      : 0x400860d3  PS      : 0x00060034  A0      : 0x80085ff0  A1      : 0x3ffb0670
A2      : 0x40085d20  A3      : 0x3ffb1c18  A4      : 0x3f801ce4  A5      : 0x40085d64
A6      : 0x3ffb197c  A7      : 0x3ffbbe58  A8      : 0x800860c5  A9      : 0x3ffb0650
A10     : 0xbad00bad  A11     : 0x3ffb0670  A12     : 0x3ffb0678  A13     : 0x00060021
A14     : 0x000000fe  A15     : 0x00000001  SAR     : 0x00000000  EXCCAUSE: 0x00000007
EXCVADDR: 0x00000000  LBEG    : 0x00000000  LEND    : 0x00000000  LCOUNT  : 0x00000000

Backtrace:0x400860d0:0x3ffb0670 0x40085fed:0x3ffb06a0 0x40086471:0x3ffb06c0 0x40084231:0x3ffb06e0 0x400848ab:0x3ffbd240 0x40085e9c:0x3ffbd260

ELF file SHA256: 003d9eb7016a05a0
```
Debug log goes here, should contain the backtrace, as well as the reset source if it is a crash.
Please copy the plain text here for us to search the error log. Or attach the complete logs but leave the main part here if the log is *too* long.
```

## Other items if possible
- [ ] sdkconfig file (attach the sdkconfig file from your project folder) 
[sdkconfig.txt](https://github.com/espressif/esp-idf/files/5678164/sdkconfig.txt)

- [ ] elf file in the ``build`` folder (**note this may contain all the code details and symbols of your project.**)
[KD_iotivity_server.elf.txt](https://github.com/espressif/esp-idf/files/5678170/KD_iotivity_server.elf.txt)

- [ ] coredump (This provides stacks of tasks.)

**Comment by Alvin1Zhang** [comment]
Thanks for reporting and letting us know, we will look into.

**Comment by YouDONG-ESP** [comment]
@ansee7221 Could you provide a complete code to reproduce this issue?

**Comment by YouDONG-ESP** [comment]
@ansee7221 We had found the cause, CONFIG_SPIRAM_ALLOW_BSS_SEG_EXTERNAL_MEMORY option will put one of our wifi variable into psram. Could you replace $IDF_PATH/esp-idf/components/esp_wifi/lib by this and see if this issue fixed? Thanks.

[lib.zip](https://github.com/espressif/esp-idf/files/5700879/lib.zip)

**Comment by ansee7221** [comment]
@YouDONG-ESP , It has been resolved after updating with the file you sent. Thank you very much.

**Comment by Alvin1Zhang** [comment]
Thanks for reporting and updates, will reopen until the fix is available, the issue will close automatically once the fix is on GitHub. Thanks.

**Comment by YouDONG-ESP** [comment]
Hi @ansee7221  it has been fixed in commit e040168a9f5697fdc68e567ecc7314cdbe61ac5b

**Comment by Alvin1Zhang** [comment]
Thanks for reporting, feel free to reopen.
